### PR TITLE
[Windows] Fix Vulkan driver crash on sub-window minimization.

### LINKED ISSUE #51537: Crash when minimising any window when a sub window is open
### Godot version

v4.0.dev.custom_build [7188cb601]

### System information

Windows 10, Vulkan Clustered, RTX 2080 SUPER (471.41)

### Issue description

Minimising either the main window or a sub window when any sub window is open causes a crash.

**Back trace**
[0] vk_optimusGetInstanceProcAddr
[1] VulkanContext::prepare_buffers (C:\Users\hle69\MHIT\GD source\godot\drivers\vulkan\vulkan_context.cpp:1761)
[2] RenderingDeviceVulkan::prepare_screen_for_drawing (C:\Users\hle69\MHIT\GD source\godot\drivers\vulkan\rendering_device_vulkan.cpp:8688)
[3] RendererCompositorRD::prepare_for_blitting_render_targets (C:\Users\hle69\MHIT\GD source\godot\servers\rendering\renderer_rd\renderer_compositor_rd.cpp:38)
[4] RendererViewport::draw_viewports (C:\Users\hle69\MHIT\GD source\godot\servers\rendering\renderer_viewport.cpp:624)
[5] RenderingServerDefault::_draw (C:\Users\hle69\MHIT\GD source\godot\servers\rendering\rendering_server_default.cpp:95)
[6] RenderingServerDefault::draw (C:\Users\hle69\MHIT\GD source\godot\servers\rendering\rendering_server_default.cpp:378)
[7] Main::iteration (C:\Users\hle69\MHIT\GD source\godot\main\main.cpp:2543)
[8] OS_Windows::run (C:\Users\hle69\MHIT\GD source\godot\platform\windows\os_windows.cpp:626)
[9] widechar_main (C:\Users\hle69\MHIT\GD source\godot\platform\windows\godot_windows.cpp:163)
[10] _main (C:\Users\hle69\MHIT\GD source\godot\platform\windows\godot_windows.cpp:185)
[11] main (C:\Users\hle69\MHIT\GD source\godot\platform\windows\godot_windows.cpp:199)
[12] __scrt_common_main_seh (d:\agent\_work\4\s\src\vctools\crt\vcstartup\src\startup\exe_common.inl:288)
[13] BaseThreadInitThunk

### Steps to reproduce

1. Open project manager
2. Click **New Project**
3. Minimise either the new project window or the main window

### Minimal reproduction project

_No response_

### PR DESCRIPTION
**Pull request by bruvzg**
Seems like NVIDIA Vulkan drivers can't create swapchain for minimized sub-window and leave it in invalid state.

This PR:
- Triggers Vulkan context resize event when window is minimized / restored.
- Adds swapchain validity checks to some Vulkan device / context methods.

Fixes #51537, fixes #54308

### CODE REVIEW

File: servers/rendering/renderer_rd/renderer_compositor_rd.cpp [hunk 1]
```diff
@@ -39,6 +39,9 @@ void RendererCompositorRD::prepare_for_blitting_render_targets() {
 
 void RendererCompositorRD::blit_render_targets_to_screen(DisplayServer::WindowID p_screen, const BlitToScreen *p_render_targets, int p_amount) {
 	RD::DrawListID draw_list = RD::get_singleton()->draw_list_begin_for_screen(p_screen);
+	if (draw_list == RD::INVALID_ID) {
+		return; // Window is minimized and do not have valid swapchain, skip drawing without printing errors.
```
Comment (akien-mga) [inline servers/rendering/renderer_rd/renderer_compositor_rd.cpp:43]:
```suggestion
		return; // Window is minimized and does not have valid swapchain, skip drawing without printing errors.
```

**Comment by KoBeWi** [comment]
I can confirm it fixes the crash.

**Review by akien-mga — APPROVED** [review]
Looks good to me, though I'd like someone from the rendering team to confirm too.

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**